.github/workflows/publish-test.yml

```diff
@@ -0,0 +1,25 @@
+name: Publish to TestPyPI
+on: workflow_dispatch  # manual invocation
+jobs:
+  run:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.10'
+      # Generate the source and binary distributions with setup.py
+      # https://packaging.python.org/tutorials/packaging-projects/#generating-distribution-archives
+      - run: python3 -m pip install --upgrade build
+      - run: python3 -m build
+      # Upload build as workflow artifact in case we need it
+      - uses: actions/upload-artifact@v3
+        with:
+          name: dist
+          path: dist/
+      # Upload the build to TestPyPI using a GitHub Action provided by PyPA
+      - uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          user: __token__
+          password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+          repository_url: https://test.pypi.org/legacy/
```

Comment:
I went to look at this action provided by PyPA, and … it's a lot of ceremony that boils down to `twine upload`. Why not just call `twine` ourselves?

Reply:
It looks official so I thought I'd use it. Though, I don't have any other experience publishing Python packages so I can change to use `twine` directly if you prefer.

Reply:
I think using `twine` directly would be more straightforward and easier to debug when/if it goes wrong. It will also naturally upgrade over time without having to configure Dependabot to monitor the workflow/action or rely on a pre-release version (e.g. `master`) of the action.

Reply:
(It also makes the workflow easier to perform locally if the need arises.)

Reply:
Fair enough, done: 8f88eab

It actually makes things run faster since we don't need to build the 3rd-party action which takes ~30s. Compare [before](https://github.com/victorlin/sphinx-theme/runs/5637184247) and [after](https://github.com/victorlin/sphinx-theme/runs/5680720751) (runs fail since you can only upload a version once).

Reply:
Oh, awesome bonus. :-)